analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -828,9 +827,283 @@ public void TryCatch_NestedThrowWithMultipleCatchFiltersOnDifferentLevels()
             validator.ValidateTagOrder(
                 "BeforeTry",
                 "InTry",
-                "InNestedTry",
-                "InNestedCatch",
-                "After");
+                "InNestedTry");
+        }
+
+        [TestMethod]
+        public void TryCatch_Throw_CatchThrown_WithVariable()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw new FormatException();
+    tag = ""UnreachableInTry"";
+}
+catch (FormatException ex)
+{
+    tag = ""InCatch"";
+}
+tag = ""End"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
```

Comment:
It would be great to validate that we are in "InCatch" with the right exception.